src/components/upload/FileUpload.js

```diff
@@ -40,43 +35,65 @@ const FileUpload = () => {
 
-    const onSubmit = () => {
-        setState("uploading");
+       const onSubmit = (e) => {
+        setOpen(true);
+        e.preventDefault();
         uploadProjectFile(projectName, projectFile)
-            .then(() => {
-                Notifications.notify("Project created.", "success")
-                setState("idle");
-            })
-            .catch((err) => {
-                Notifications.notify(`Failed to create project.\n${err}`, "error")
-                setState("idle");
-            })
+         if (projectName && projectFile) {
+            Notifications.notify("Project created.", "success")
+            setState("idle");
+            setOpen(true);
+            navigate("/generate", { replace: true });
+        } else if (!projectName && projectFile){
+            Notifications.notify(`Failed to create project.\n`, "error")
+            setState("idle"); 
+            alert('please enter the project name');
+        } else if (projectName && !projectFile){
+          Notifications.notify(`Failed to create project.\n`, "error")
+          setState("idle"); 
+          alert('please upload the project file');
+      }  else {
+           Notifications.notify(`Failed to create project.\n`, "error")
+           setState("idle"); 
+           alert('please upload the project file and enter the project name');
+      }
+      };
+      const handleInputChange = (e) => {
+        setProjectName(e.target.value);
+        enableButton(e.target.value);
     };
-
     return (
         <FileUploadRoot>
             <FileDropZone
                 file={projectFile}
                 onFile={setProjectFile}
             />
-
+                
             <Confirmation>
-                <Input
-                    placeholder="Project name"
+               <Box sx={{ '& > :not(style)': { m: 1, width: '25ch' },  }}>
+                <TextField label="Project name" variant="standard" placeholder="Project name"
                     value={projectName}
-                    onChange={(e) => setProjectName(e.target.value)}
-                />
-
-                <UploadButton
-                    onClick={onSubmit}
-                    disabled={!projectName || !projectFile || state === "uploading"}
-                >
-                    Upload file
-                </UploadButton>
+                    onChange={handleInputChange}>
+                     
+                </TextField>
+                <Button variant="contained" onClick={onSubmit}  endIcon={<UploadIcon />} >
+                  Upload file        
+                </Button>
+                <div>
+                <Projectdecription
+                       id="outlined-multiline-static"
+                       label="project decription"
+                       multiline
+                       rows={4}/>
+                </div>
+                </Box>
             </Confirmation>
         </FileUploadRoot>
     );
 };
 
-export default FileUpload;
+export default FileUpload;
```

Comment:
```suggestion
export default FileUpload;

```
Keep the newline. We need exactly one newline.